File: app/src/main/java/com/theah64/dotsexample/MainActivity.java
package com.theah64.dotsexample;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;

import com.theah64.dots.Dots;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    private Dots dots;
    private int currentIndex = 0;
    private final int totalDots = 8;

    public static void start(Context context) {
        context.startActivity(new Intent(context, MainActivity.class));
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        this.dots = findViewById(R.id.dots);
        dots.setCount(totalDots);

        findViewById(R.id.bMoveBack).setOnClickListener(this);
        findViewById(R.id.bMoveForward).setOnClickListener(this);

    }


    @Override
    public void onClick(View view) {


        switch (view.getId()) {

            case R.id.bMoveBack:
                dots.moveBack();
                break;

            case R.id.bMoveForward:
                dots.moveForward();
                break;
        }
    }
}

File: app/src/main/java/com/theah64/dotsexample/MainActivity.kt
package com.theah64.dotsexample

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.View

import com.theah64.dots.Dots

class MainActivity : AppCompatActivity(), View.OnClickListener {

    private var dots: Dots? = null
    private val currentIndex = 0
    private val totalDots = 8

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val toolbar = findViewById<View>(R.id.toolbar) as Toolbar
        setSupportActionBar(toolbar)

        this.dots = findViewById(R.id.dots)
        dots!!.setCount(totalDots)

        findViewById<View>(R.id.bMoveBack).setOnClickListener(this)
        findViewById<View>(R.id.bMoveForward).setOnClickListener(this)

    }


    override fun onClick(view: View) {


        when (view.id) {

            R.id.bMoveBack -> dots!!.moveBack()

            R.id.bMoveForward -> dots!!.moveForward()
        }
    }

    companion object {

        fun start(context: Context) {
            context.startActivity(Intent(context, MainActivity::class.java))
        }
    }
}
